src/main/java/com/jagrosh/giveawaybot/commands/SettingsCommand.java

```diff
@@ -32,6 +39,8 @@
 public class SettingsCommand extends Command
 {
     private final Bot bot;
+    private final String EMOTE_REGEX = "<a?:([a-zA-Z0-9_]+):([0-9]+)>";
```

Comment:
Use a different regex to take advantage of character classes and respect character limits - an emoji's name ranges between 2 and 32 characters, and snowflakes are 17 to 19 characters, at least until 2090 :shipit: :

```suggestion
    private final String EMOTE_REGEX = "<a?:(\w{2,32}):(\d{17,19})>";
```

Reply:
This is what JDA uses internally just extracted for convenience.